src/test/java/picard/illumina/parser/TileMetricsUtilTest.java

```diff
@@ -8,67 +8,90 @@
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.List;
+import java.util.Random;
 import java.util.stream.IntStream;
 
 public class TileMetricsUtilTest {
     @Test
     public void testFindTileMetrics() throws IOException {
-        Path baseDir = createTestDirs(0, true);
-        File tileMetricsFile = TileMetricsUtil.renderTileMetricsFileFromBasecallingDirectory(
+        List<Integer> populatedCycleDirs = new ArrayList<>();
+        Path baseDir = createTestDirs(0, populatedCycleDirs, true);
+        List<File> tileMetricsFiles = TileMetricsUtil.findTileMetricsFiles(
                 baseDir.toFile(), 0, false);
 
-        Assert.assertEquals(tileMetricsFile.getAbsolutePath(),
-                baseDir.resolve(TileMetricsUtil.INTEROP_SUBDIRECTORY_NAME)
-                        .resolve(TileMetricsUtil.TILE_METRICS_OUT_FILE_NAME).toString());
-
+        List<File> expected = generateFindTileMetricsExpected(baseDir, populatedCycleDirs, true);
+        Assert.assertEquals(tileMetricsFiles, expected);
         IOUtil.deleteDirectoryTree(baseDir.toFile());
     }
 
     @Test(expectedExceptions = IllegalStateException.class)
```

Comment:
If this test throws an exception, the `deleteDirectoryTree()` operation at the end will never execute.

Not sure what the best solution here is, using a shared test directory that the class creates and then deletes, using an `@AfterClass` method, might be best.